shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -249,8 +249,37 @@ Engine::Engine(Delegate& delegate,
            external_view_embedder = GetExternalViewEmbedder(),
            vsync_offset = product_config.get_vsync_offset(),
            vsync_handle = vsync_event_.get(),
-           keyboard_listener_request = std::move(keyboard_listener_request)](
-              flutter::Shell& shell) mutable {
+           keyboard_listener_request = std::move(keyboard_listener_request),
+           &product_config](flutter::Shell& shell) mutable {
+            OnShaderWarmup on_shader_warmup = nullptr;
+            if (product_config.enable_shader_warmup()) {
```

Comment:
Are there bugs for removing these 2 flags once shader warmup is triggered from the dart code?

If the Dart code can choose to trigger it, the runtime flags are adding extra complexity and noise so we should remove them.

Reply:
I was planning to leave them in so that we could kill the shader warmup at runtime if we wanted to for debugging purposes (even long term) but I can remove them if you feel strongly about it. Personally I lean towards lots of runtime flags being good for debugging purposes but that is kindof predicated on the build system being able to handle lots of flags without exponentially growing complexity which i know is not the case. 

I'll plan on removing them for now but maybe we should discuss whether runtime flags are intended to be transient or permanent

Reply:
actually looking closer I'm only gonna keep `enable_shader_warmup`, I was gonna delete `enable_shader_warmup_dart_hooks` and all the code paths which depend on it once this sticks. I was planning on leaving a runtime toggle for shader warmup but i dont feel terribly strongly about it.

Reply:
I'm just wondering if we even need the runtime toggle in the runner at all once it's completely client-triggered.  At that point, if we want to disable shader warmup we can do it in the client code by not sending the warmup message.  This means less code and overall less complexity in the runner itself.

There is another reason to avoid these flags and that is because this product configuration is system-wide and baked into the system image.  If you want one flutter app to disable shader warmup but have another enable it, that's impossible to do with the current config flag.   There's a similar concern around tests too -- since the flag is system-wide it's really hard to write a test case for flag off and a second test case w/ flag on.

However what the config flag does allow is for product owners to force shader warmup to OFF if these don't want any flutter apps using it for some reason.  Do we want to support that use-case?

Reply:
Yeah that rationale makes sense to me. I think what we probably want post migration is a runtime flag on each client that lets you tell that client to stop doing shader warmup, rather than telling the runner itself to stop doing shader warmup. I can look into that, I don't know how the runtime flags work there right now.